server/platform/shared/filestore/s3store.go

```diff
@@ -88,8 +88,17 @@ func (s *S3FileBackendNoBucketError) Error() string {
 	return "no such bucket"
 }
 
-// NewS3FileBackend returns an instance of an S3FileBackend.
+// NewS3FileBackend returns an instance of an S3FileBackend and determine if we are in Mattermost cloud or not.
 func NewS3FileBackend(settings FileBackendSettings) (*S3FileBackend, error) {
+	return newS3FileBackend(settings, os.Getenv("MM_CLOUD_FILESTORE_BIFROST") != "")
+}
+
+// NewS3FileBackendWithoutBifrost returns an instance of an S3FileBackend assuming we are not on cloud.
```

Comment:
"assuming we are not in cloud" is not really correct. We _are_ still in cloud, but just don't want to use bifrost.